homeassistant/components/telegram_bot/__init__.py

```diff
@@ -756,16 +756,15 @@ async def send_message(self, message="", target=None, context=None, **kwargs):
                 message_thread_id=params[ATTR_MESSAGE_THREAD_ID],
                 context=context,
             )
-            if msg is not None:
-                msg_ids[chat_id] = msg.id
+            msg_ids[chat_id] = msg.id
         return msg_ids
 
     async def delete_message(self, chat_id=None, context=None, **kwargs):
         """Delete a previously sent message."""
         chat_id = self._get_target_chat_ids(chat_id)[0]
         message_id, _ = self._get_msg_ids(kwargs, chat_id)
         _LOGGER.debug("Delete message %s in chat ID %s", message_id, chat_id)
-        deleted = await self._send_msg(
+        await self._send_msg(
```

Comment:
A nice followup PR for this integration would be to add type hints, that would increase the maintainability by a lot